src/app/edit-maprule/edit-maprule.component.ts

```diff
@@ -0,0 +1,153 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output } from '@angular/core';
+import { Validators, FormGroup, FormBuilder, FormArray, FormControl } from '@angular/forms';
+
+import { FieldConfig } from '../shared/interfaces/field-config.interface';
+import { PresetComponent } from './preset/preset.component';
+import { DisabledFeatureComponent} from './disabled-feature/disabled-feature.component'
+import { MapRulesService } from '../core/services/maprules.service';
+import { NavigationService } from '../core/services/navigation.service';
+import { Router, ActivatedRoute } from '@angular/router';
+import { FieldConfigService } from 'src/app/core/services/field-config.service';
+import { TagInfoService } from '../core/services/tag-info.service';
+import { fbind } from 'q';
+
+@Component({
+  exportAs: 'edit-maprule',
+  selector: 'app-edit-maprule',
+  styleUrls: [
+    '../shared/components/content.group.css',
+    './edit-maprule.component.css'
+  ],
+  templateUrl: './edit-maprule.html'
+})
+export class EditMapRuleComponent implements OnInit {
+  @Output()
+  submit: EventEmitter<any> = new EventEmitter<any>();
+  form: FormGroup;
+
+  presetGeometries = [];
+
+  get presets() { return this.form.get('presets') as FormArray; }
+  get disabledFeatures() {
+    return this.form.get('disabledFeatures') as FormArray;
+  }
+
+  constructor(
+    private fb: FormBuilder,
+    private maprules: MapRulesService,
+    private tagInfoSevice: TagInfoService,
+    private nav: NavigationService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private fieldConfig: FieldConfigService
+  ) {}
+
+  ngOnInit() {
+    this.route.params.subscribe(
```

Comment:
Subscribes to changes to the router when we are behind `/edit` and...

1. when we have an id, when its and existing maprules, initializes the form w/external maprule
2. otherwise, creates a blank one.

both cases also emit tot he fieldConfig event handler that we've initialized. This event is head by the disabled features and preset components and used to render those uis as open/closed.